### PR TITLE
Maintain the same interface when using READABLE_STREAM=disable

### PR DESCRIPTION
**Pull request by mcollina**
This would need to be backported to the 1.0 and 1.1 lines, as they suffer from the same bug.

Fixes https://github.com/nodejs/node/issues/12006

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
you probably want to build this on top of #260

**Comment by mcollina** [comment]
@calvinmetcalf updated and refactored.

I'm wondering if we should change this replacement as well for browsers:
https://github.com/nodejs/readable-stream/blob/master/lib/_stream_readable.js#L27-L36.

and flag `"stream": false`  in the  `"browser"`  field in `package.json`?

**Comment by calvinmetcalf** [comment]
yes good call

File: transform.js [hunk 1]
```diff
@@ -1 +1 @@
-module.exports = require("./lib/_stream_transform.js")
+module.exports = require('./readable').Transform
```
Comment (calvinmetcalf) [inline transform.js]: could this theoretically pull in more then it did previously? probably just passthrough right

**Comment by mcollina** [comment]
@nodejs/streams may I get some LGTMs so we can release this?

I think this is a patch release, as we are fixing a bunch of things.

**Pull request merged**